src/main/java/com/cardinity/validators/PaymentValidator.java

```diff
@@ -55,6 +55,10 @@ else if (payment.getPaymentInstrument().getType() == Payment.Method.RECURRING)
         if (payment.getThreeds2Data() != null) {
             THREEDS_2_DATA_VALIDATOR.validate(payment.getThreeds2Data());
         }
+
+        if (payment.getStatementDescriptorSuffix() != null &&
+                !ValidationUtils.validateIntervalLength(payment.getStatementDescriptorSuffix(), 0,6))
```

Comment:
Currently 25 is total allowed length of the descriptor. Allowed suffix length depends on the base descriptor length. So if base part is short, suffix can be longer. It also differs for each merchant. So the best we can do is to restrict maximum theoretical length of the suffix. So lets check max length 20 characters.

Please also review code formatting here as it is a bit off

Reply:
Resolved
https://github.com/cardinity/cardinity-sdk-java/pull/3/commits/5ff6e03cdbe6b5676919a77ff9410a38078739ff